src/components/views/admin/TableView.tsx

```diff
@@ -0,0 +1,169 @@
+  TABLE_VIEW_USER_REVIEW_DIALOG_CYPRESS,
+  TABLE_VIEW_VIEW_COMMENTS_CELL_CYPRESS,
+  tableRowUserCypress,
+} from '../../../config/selectors';
+import { CommentType } from '../../../interfaces/comment';
+import { getOrphans } from '../../../utils/comments';
+import CodeReviewWrapper from '../../common/CodeReviewWrapper';
+import CustomDialog from '../../common/CustomDialog';
+import {
+  AppDataProvider,
+  useAppDataContext,
+} from '../../context/AppDataContext';
+import { useMembersContext } from '../../context/MembersContext';
+import OrphanComments from './OrphanComments';
+
+const DEFAULT_CURRENT_USER = {
+  name: '',
+  id: '',
+};
+
+const TableView: FC = () => {
+  const { t } = useTranslation();
+  const [openCommentView, setOpenCommentView] = useState(false);
+  const [currentUser, setCurrentUser] = useState(DEFAULT_CURRENT_USER);
+  const members = useMembersContext();
+  const { appData } = useAppDataContext();
+
+  const comments = appData?.filter(
+    (res) => res.type === APP_DATA_TYPES.COMMENT,
+  ) as List<CommentType>;
+
+  const renderTableBody = (): ReactElement[] | ReactElement | null => {
+    const orphansId = getOrphans(comments).map((c) => c.id);
+    const nonOrphanComments = comments?.filter(
+      (c) => !orphansId.includes(c.id),
+    );
+    if (nonOrphanComments?.isEmpty()) {
+      // show that there are no comments available
+      return (
+        <TableRow>
+          <TableCell
+            colSpan={NB_COL_TABLE_VIEW_TABLE}
+            data-cy={TABLE_NO_COMMENTS_CYPRESS}
+          >
+            {t('No Comments')}
+          </TableCell>
+        </TableRow>
+      );
+    }
+    const commentsByUsers = nonOrphanComments
+      ?.groupBy(({ memberId }) => memberId)
+      .toArray();
+    return (
+      commentsByUsers?.map(([userId, userComments]) => {
+        const userName =
+          members?.find(({ id }) => id === userId)?.name || ANONYMOUS_USER;
+        return (
+          <TableRow key={userId} data-cy={tableRowUserCypress(userId)}>
+            <TableCell data-cy={TABLE_VIEW_USERNAME_CELL_CYPRESS}>
+              {userName}
+            </TableCell>
+            <TableCell data-cy={TABLE_VIEW_HELP_NEEDED_CELL_CYPRESS}>
+              {false}
+            </TableCell>
+            <TableCell data-cy={TABLE_VIEW_NB_COMMENTS_CELL_CYPRESS}>
+              <div>{userComments.count()}</div>
+            </TableCell>
+            <TableCell data-cy={TABLE_VIEW_VIEW_COMMENTS_CELL_CYPRESS}>
+              <IconButton
+                data-cy={TABLE_VIEW_OPEN_REVIEW_BUTTON_CYPRESS}
+                onClick={() => {
+                  setCurrentUser({
+                    name: userName,
+                    id: userId,
+                  });
+                  setOpenCommentView(true);
+                }}
+              >
+                <InputIcon color="primary" />
+              </IconButton>
+            </TableCell>
+          </TableRow>
+        );
+      }) || null
```

Comment:
Not sure why this is here.

Reply:
It was in case there was no comments to display. But i refactored the logic to make sure that there is an early return in cas there is nothing to show, and then we can be sure that the lists are not empty so we can remove that.